### PR TITLE
feat: Add git root discovery for default base path

### PR DESCRIPTION
**Pull request by osterman**
## Summary

- Enables running Atmos from any subdirectory like Git automatically discovers repository root
- Local configuration always takes precedence over git root discovery  
- Feature controlled by ATMOS_GIT_ROOT_BASEPATH environment variable

## Test Plan

- [x] All unit tests passing (18 new test cases in git_root_basepath_test.go)
- [x] Existing CLI tests passing with ATMOS_GIT_ROOT_BASEPATH=false set
- [x] Build successful with `make build`
- [x] Code follows architectural patterns and linting rules

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Automatic Git repository root discovery: Atmos can run from subdirectories and use the repo root as the base path.

* **Behavior**
  * Local config files/directories in the current directory take precedence; discovery applies only when no local config is present.
  * Discovery can be disabled via an environment variable for tests or opt-out.

* **Documentation**
  * New docs and blog post with examples, rollout guidance, and migration notes.

* **Tests**
  * Added tests covering discovery, local-config detection, env override, and edge cases.

* **Other**
  * Additional trace logs to aid debugging and rollout validation.

### CODE REVIEW

File: pkg/config/git_root.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+//
+// This allows users to run `atmos` from anywhere in a Git repository
+// without needing to specify --config-dir or base_path.
+func applyGitRootBasePath(atmosConfig *schema.AtmosConfiguration) error {
+	// Allow tests to disable git root discovery.
+	//nolint:forbidigo // ATMOS_GIT_ROOT_BASEPATH is specifically for feature control, not application configuration.
+	if os.Getenv("ATMOS_GIT_ROOT_BASEPATH") == "false" {
+		log.Trace("Git root base path disabled via ATMOS_GIT_ROOT_BASEPATH=false")
+		return nil
+	}
```
[the remark this answers is not in the document]
Comment (aknysh) [reply pkg/config/git_root.go]: This is a bootstrap configuration variable that must be read during LoadConfig() (line 105 of load.go), before processEnvVars() is called (line 163 of config.go). It follows the same pattern as ATMOS_CLI_CONFIG_PATH (load.go:279), which 
  also reads directly from os.Getenv() because it controls the config loading process itself. Moving it to Settings would create a circular dependency since we need this value to determine how to load the config, but Settings is populated after config loading completes.
